examples/browser-webrtc/client/src/pinger.rs

```diff
@@ -0,0 +1,79 @@
+use super::fetch_server_addr;
+use async_channel::Sender;
+use futures::StreamExt;
+use libp2p::core::Multiaddr;
+use libp2p::identity;
+use libp2p::identity::PeerId;
+use libp2p::ping;
+use libp2p::swarm::{keep_alive, NetworkBehaviour, SwarmBuilder, SwarmEvent};
+use libp2p::{multiaddr, swarm};
+use std::convert::From;
+
+pub async fn start_pinger(sendr: Sender<f32>) -> Result<(), PingerError> {
+    let addr_fut = fetch_server_addr();
+
+    let local_key = identity::Keypair::generate_ed25519();
+    let local_peer_id = PeerId::from(local_key.public());
+
+    let mut swarm = SwarmBuilder::with_wasm_executor(
+        libp2p_webrtc_websys::Transport::new(libp2p_webrtc_websys::Config::new(&local_key)).boxed(),
+        Behaviour {
+            ping: ping::Behaviour::new(ping::Config::new()),
+            keep_alive: keep_alive::Behaviour,
+        },
+        local_peer_id,
+    )
+    .build();
+
+    log::info!("Running pinger with peer_id: {}", swarm.local_peer_id());
+
+    let addr = addr_fut.await;
```

Comment:
Just mee being fancy with the async call I suppose. Technically I can run the swarm code above it asynchronously while I wait for the server to respond instead of blocking on the server response. We are talking about milliseconds (maybe microseconds) so it's not a big deal, just the way my mind thinks about async coding. I can change it if it makes it more clear.

Reply:
Futures are lazy in Rust so nothing actually happens until you await them (apart from the memory allocations needed to hold their state).

Reply:
Fixed